9.5/root/usr/share/container-scripts/postgresql/common.sh

```diff
@@ -55,8 +55,17 @@ EOF
   exit 1
 }
 
+check_cred_secret() {
+    local credpath="$1"
+    [ -f "$credpath/username" ] && \
+    [ -f "$credpath/password" ] && \
+    [[ "$(<"$credpath/username")" =~ $psql_identifier_regex ]] && \
+    [[ "$(<"$credpath/password")" =~ $psql_password_regex ]] &&
+    [ "$(wc -c < "$credpath/username")" -le 63 ]
```

Comment:
I don't think that we need to bother doing the regexps, but I understand that this is sort of guideline.  Reason: That's just re-implementing of the checks which are done by PostgreSQL itself.  I bet this check exists in OpenShift gui, too.